plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,302 @@
+				$final_status   = 'recommended';
+				$fail_details[] = array(
+					'filename' => $filename,
+					'reason'   => __( 'No far-future headers but conditionally cached', 'performance-lab' ),
+				);
+			}
+		} else {
+			// If there's a max-age or expires but below threshold, we skip conditional.
+			$final_status   = 'recommended';
+			$fail_details[] = array(
+				'filename' => $filename,
+				'reason'   => $check['reason'],
+			);
+		}
+	}
+
+	return array(
+		'final_status' => $final_status,
+		'details'      => $fail_details,
+	);
+}
+
+/**
+ * Checks if far-future expiration headers are enabled.
+ *
+ * @since n.e.x.t
+ *
+ * @param WpOrg\Requests\Utility\CaseInsensitiveDictionary|array<string, string|array<string>> $headers Response headers.
+ * @return array{passed: bool, reason: string}|false Detailed result. If passed=false, reason explains why it failed and false if no headers found.
+ */
+function perflab_ffh_check_headers( $headers ) {
+	/**
+	 * Filters the threshold for far-future headers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $threshold Threshold in seconds.
+	 */
+	$threshold = apply_filters( 'perflab_far_future_headers_threshold', YEAR_IN_SECONDS );
+
+	$cache_control = isset( $headers['cache-control'] ) ? $headers['cache-control'] : '';
+	$expires       = isset( $headers['expires'] ) ? $headers['expires'] : '';
+
+	// Check Cache-Control header for max-age.
+	$max_age = 0;
+	if ( '' !== $cache_control ) {
+		// There can be multiple cache-control headers, we only care about max-age.
+		$controls = is_array( $cache_control ) ? $cache_control : array( $cache_control );
+		foreach ( $controls as $control ) {
+			if ( (bool) preg_match( '/max-age\s*=\s*(\d+)/', $control, $matches ) ) {
+				$max_age = (int) $matches[1];
+				break;
+			}
+		}
+	}
+
+	// If max-age meets or exceeds the threshold, we consider it good.
+	if ( $max_age >= $threshold ) {
+		return array(
+			'passed' => true,
+			'reason' => '',
+		);
+	}
+
+	// If max-age is too low or not present, check Expires.
+	if ( is_string( $expires ) && '' !== $expires ) {
+		$expires_time = strtotime( $expires );
+		if ( (bool) $expires_time && ( $expires_time - time() ) >= $threshold ) {
+			// Good - Expires far in the future.
+			return array(
+				'passed' => true,
+				'reason' => '',
+			);
+		}
+
+		// Expires header exists but not far enough in the future.
+		if ( $max_age > 0 && $max_age < $threshold ) {
+			return array(
+				'passed' => false,
+				'reason' => __( 'max-age below threshold', 'performance-lab' ),
+			);
+		}
+		return array(
+			'passed' => false,
+			'reason' => __( 'expires below threshold', 'performance-lab' ),
+		);
+	}
+
+	// No max-age or expires found at all or max-age < threshold and no expires.
+	if ( 0 === $max_age ) {
+		return false;
+	} else {
+		// max-age was present but below threshold and no expires.
+		return array(
+			'passed' => false,
+			'reason' => __( 'max-age below threshold', 'performance-lab' ),
+		);
+	}
```

Comment:
It would seem helpful to indicate the `max-age` and `expires` values in the error message so you can see what the actual values are. Likewise, shouldn't the threshold be added to the error message so the user knows the minimum TTL that they should configure the expires and max-age to be?

Reply:
I think converting the expires to seconds makes sense. I don't think we need to worry about converting to anything other than seconds, however. Just pass the number of seconds through `number_format_i18n()` to get some better formatting.